website/docs/indexes.md

```diff
@@ -0,0 +1,126 @@
+---
+sidebar_position: 7
+---
+
+# Indexes
+
+Indexes are essential in improving query performance by enabling fast retrieval of relevant documents when querying large collections.
+
+Indexes in FerretDB are created based on a specific field(s) within documents.
+When an index is created, this creates a data structure that maps between the index values and the document locations, making it possible to retrieve documents more quickly based on those fields.
```

Comment:
`this creates a data structure that maps between the index values and the document locations`
This sounds a bit vague and specific at the same time, it doesn't give much information to the users though.

Reply:
@Fashander any thoughts about this comment?

Reply:
I've updated the sentence to be a bit clearer. Although maybe some graphical representation will be better in explaining it